src/reference_tree.ml

```diff
@@ -553,6 +556,146 @@ let get_ceil_data f reftree path =
         | _, [] -> d
     in aux None [] path
 
+(** Add alternative validation function to be called on an existing tree.
+    This allows folding over tree for full validation.
+    Numbered comments list constraints as described above validate_path.
+ *)
+let validate_tree_at_path validators_dir rt ct path value =
+    if Util.is_empty path then ()
+    else
+    let show_path p =
+        Printf.sprintf "[%s]" @@ Util.string_of_list p
+    in
+    let refp = refpath rt path in
+    (* 6. It's a node that is neither leaf nor tag value with a name that
+          doesn't exist in the reference tree
+     *)
+    if not (Vytree.exists rt refp) then
+        let msg = Printf.sprintf "Path %s is not in reference tree\n" (show_path path)
+        in raise (Validation_error msg)
+    else
+    let node =
+        try
+            Vytree.get ct path
+        with Vytree.Nonexistent_path ->
+            let msg = Printf.sprintf "Path %s is not in config file\n" (show_path path)
```

Comment:
I have a preference for `Printf.ksprintf` and a helper function like `let validation_error msg = raise (Validation_error msg)`. It's just a preference, of course.

Reply:
Printf.ksprintf is a useful suggestion: revision/improvement of error messages will follow in a separate PR.